core/server/worker/src/main/java/alluxio/worker/file/DefaultFileSystemWorker.java

```diff
@@ -170,12 +171,22 @@ public void start() {
 
   @Override
   public void stop() {
+    // Steps to shutdown:
+    // 1. Gracefully shutting down the runnables running in the executors first.
+    // 2. Shutdown the executors.
+    // 3. Shutdown the clients. This needs to happen after the executors is shutdown because
+    //    runnables running in the executors might be using the clients.
     mSessionCleaner.stop();
     if (mFilePersistenceService != null) {
       mFilePersistenceService.cancel(true);
     }
-    mFileSystemMasterWorkerClient.close();
     // This needs to be shutdownNow because heartbeat threads will only stop when interrupted.
     getExecutorService().shutdownNow();
+    try {
+      getExecutorService().awaitTermination(10, TimeUnit.MINUTES);
```

Comment:
Why 10 minutes? I would have expected something more like 5-10 seconds. Does this mean that it could take 10 minutes for a standby to upgrade to leader?